ee/tabby-webserver/src/cron/scheduler.rs

```diff
@@ -0,0 +1,89 @@
+use std::{process::Stdio, sync::Arc, time::Duration};
+
+use tokio::io::AsyncBufReadExt;
+use tokio_cron_scheduler::Job;
+use tracing::{info, warn};
+
+use crate::schema::{job::JobService, worker::WorkerService, ServiceLocator};
+
+pub async fn scheduler_job(
```

Comment:
This function is long and does many things. I think it should be split up, particularly building the command and piping process output.

Reply:
ack, should be better to do as a followup pr